### PR TITLE
[type: refactor] Admin distributed lock by spring-integration-jdbc

### PR DESCRIPTION
**Pull request by crudboy**
Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

About #5448 Task: Admin distributed lock by spring-integration-mysql

### CODE REVIEW

**Comment by moremind** [comment]
pls add a table for all SQL

**Comment by crudboy** [comment]
Doesn't seem to support jkd19,20.21?

**Comment by crudboy** [comment]
It seems that some files do not support jkd19,20,21?

**Comment by crudboy** [comment]
@moremind Could you please take a look  and tell if there any changes needed or any mistakes I need to fix?

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/AbstractContextPathRegisterService.java [hunk 1]
```diff
@@ -17,37 +17,47 @@
...
         try {
+            String contextPathSelectorId = getSelectorService().registerDefault(dto, name, "");
+            // avoid repeated registration for many client threads
+            // many client threads may register the same context path for contextPath plugin at the same time
             if (Objects.nonNull(getRuleService().findBySelectorIdAndName(contextPathSelectorId, dto.getContextPath()))) {
                 return;
             }
             ContextMappingRuleHandle handle = new ContextMappingRuleHandle();
-            handle.setContextPath(PathUtils.decoratorContextPath(dto.getContextPath()));
+            handle.setContextPath(contextPath);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/AbstractContextPathRegisterService.java:60]: why change this?

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/AbstractContextPathRegisterService.java (hunk 1 above)
Comment (crudboy) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/AbstractContextPathRegisterService.java:60]: I didn't change contextPath.  use contextPath in  registry#obtain method parameters. 
```java
        String contextPath = PathUtils.decoratorContextPath(dto.getContextPath());
        String key = LOCK_KEY_PREFIX + contextPath;
        Lock lock = registry.obtain(key);
       .....
       handle.setContextPath(contextPath);
```

**Comment by moremind** [comment]
LGTM

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/AbstractContextPathRegisterService.java [hunk 2]
```diff
@@ -17,37 +17,47 @@
...
-        // avoid repeated registration for many client threads
-        // many client threads may register the same context path for contextPath plugin at the same time
-        RegisterExecutionLock lock = registerExecutionRepository.getLock(name);
+        String contextPath = PathUtils.decoratorContextPath(dto.getContextPath());
+        String key = LOCK_KEY_PREFIX + contextPath;
+        Lock lock = registry.obtain(key);
+        if (logger.isDebugEnabled()) {
+            logger.debug("register context-path obtain lock for key:{}", key);
+        }
         lock.lock();
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/AbstractContextPathRegisterService.java:51]: put it in try

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/AbstractContextPathRegisterService.java (hunk 2 above)
Comment (crudboy) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/AbstractContextPathRegisterService.java]: change lock.lock() put it in try

**Review by moremind — APPROVED** [review]

**Comment by moremind** [comment]
LGTM

**Comment by crudboy** [comment]
@moremind Excuse me. Could you please reapproving ？  i  fix ci

File: shenyu-admin/src/test/java/org/apache/shenyu/admin/service/register/AbstractContextPathRegisterServiceTest.java [hunk 3]
```diff
@@ -55,20 +58,56 @@ class AbstractContextPathRegisterServiceTest {
...
     public void testRegisterContextPath() {
         MetaDataRegisterDTO dto = MetaDataRegisterDTO.builder().build();
         dto.setContextPath("Context_Path");
         dto.setAddPrefixed(true);
-        RegisterExecutionLock registerExecutionLock = mock(RegisterExecutionLock.class);
         when(selectorService.registerDefault(dto, PluginEnum.CONTEXT_PATH.getName(), "")).thenReturn("Context_Path_Selector_Id");
-        when(registerExecutionRepository.getLock(any())).thenReturn(registerExecutionLock);
+        // org.springframework.integration.jdbc.lock.JdbcLockRegistry.JdbcLock is private and cannot be mocked directly so we mock the LockRegistry and return a mock MockLock
+        // here  mock ReentrantLock   cause cpu usage 100% in the jdk 19 20 21 environment
+        when(registry.obtain(any())).thenReturn(mock(MockLock.class));
```
Comment (crudboy) [inline shenyu-admin/src/test/java/org/apache/shenyu/admin/service/register/AbstractContextPathRegisterServiceTest.java:71]: i change this mock ReentrantLock   cause cpu usage 100% in the jdk 19 20 21 environment

**Comment by crudboy** [comment]
i change this file [AbstractContextPathRegisterServiceTest](https://github.com/apache/shenyu/pull/5457/files#r1513828406)

**Comment by moremind** [comment]
> @moremind Could you please take a look  and tell if there any changes needed or any mistakes I need to fix?

no

**Review by yunlongn — APPROVED** [review]

**Pull request merged**